pkg/cmd/cli/cmd/login.go

```diff
@@ -102,7 +102,7 @@ func (o *LoginOptions) Complete(f *osclientcmd.Factory, cmd *cobra.Command, args
 		o.StartingKubeConfig = kclientcmdapi.NewConfig()
 	}
 
-	addr := flagtypes.Addr{Value: "localhost:8443", DefaultScheme: "https", DefaultPort: 8443, AllowPrefix: true}.Default()
+	addr := flagtypes.Addr{Value: "localhost:8443", DefaultScheme: "https", AllowPrefix: true}.Default()
```

Comment:
This is what fixes https://github.com/openshift/origin/issues/9570. With `DefaultPort: 8443`, in case the port wasn't specified it would use `8443` instead of `443`, the default port for the given scheme.